src/System.Xml.XmlDocument/src/System/Xml/Dom/XmlElement.cs

```diff
@@ -58,8 +58,14 @@ public override XmlNode CloneNode(bool deep)
                 foreach (XmlAttribute attr in Attributes)
                 {
                     XmlAttribute newAttr = (XmlAttribute)(attr.CloneNode(true));
-                    if (attr is XmlUnspecifiedAttribute && attr.Specified == false)
-                        ((XmlUnspecifiedAttribute)newAttr).SetSpecified(false);
+                    if (attr.Specified == false)
```

Comment:
@stephentoub I've looked implementation of these classes. Property `Specified` introduced in `XmlAttribute` just returns `true` and not overridden in derived classes except `XmlUnspecifiedAttribute` where it overridden and can be `true` or `false`.